src/java.base/share/classes/jdk/internal/vm/ThreadSnapshot.java

```diff
@@ -48,6 +48,9 @@ class ThreadSnapshot {
     // set by ThreadSnapshot.of()
     private ThreadBlocker blocker;
 
+    // the owner of the blockerObject when the object is park blocker and is AQS
+    private Thread parkBlockerOwner;
```

Comment:
Just wondering whether it makes sense to make the owner field part of the `ThreadBlocker` record?

Reply:
The parkBlocker is a property of the thread. When the parkBlocker is an AbstractOwnableSynchronizer then the owner is a property of that park blocker type. So yes, in modelling terms it would be more accurate. In practical terms it doesn't matter as it's all encapsulated but there is likely more work in this area so it would be better to get it right.